### PR TITLE
Multisigner: keys holder redundancy support

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added redundancy support for keysHolder

### CODE REVIEW

File: keysManagement/errors.go [hunk 1]
```diff
@@ -6,3 +6,4 @@ var errDuplicatedKey = errors.New("duplicated key found")
 var errMissingPublicKeyDefinition = errors.New("missing public key definition")
 var errNilKeyGenerator = errors.New("nil key generator")
 var errNilP2PIdentityGenerator = errors.New("nil p2p identity generator")
+var errInvalidValue = errors.New("invalid value")
```
Comment (bogdan-rosianu) [inline keysManagement/errors.go:9]: could have grouped them in a `var (  )` group

File: keysManagement/keysHolder.go [hunk 2]
```diff
@@ -153,6 +161,89 @@ func (holder *virtualPeersHolder) GetMachineID(pkBytes []byte) (string, error) {
...
+			errMissingPublicKeyDefinition, hex.EncodeToString(pkBytes))
+	}
+
+	pInfo.resetRoundsWithoutReceivedMessages()
+
+	return nil
+}
+
+// GetAllManagedKeys returns all keys that will be managed by this node
+func (holder *virtualPeersHolder) GetAllManagedKeys() map[string]crypto.PrivateKey {
```
Comment (bogdan-rosianu) [inline keysManagement/keysHolder.go:199]: the function name includes an `All` but you filter the output. Might rename?

File: keysManagement/keysHolder.go [hunk 3]
```diff
@@ -153,6 +161,89 @@ func (holder *virtualPeersHolder) GetMachineID(pkBytes []byte) (string, error) {
...
+	holder.mut.RLock()
+	defer holder.mut.RUnlock()
+
+	allManagedKeys := make(map[string]crypto.PrivateKey)
+	for pk, pInfo := range holder.data {
```
Comment (bogdan-rosianu) [inline keysManagement/keysHolder.go:204]: `holder.isMainMachine` can be extracted in a var outside the for loop - slight improvement

File: keysManagement/errors.go (hunk 1 above)
Comment (iulianpascalau) [reply keysManagement/errors.go:9]: done

File: keysManagement/keysHolder.go (hunk 3 above)
Comment (iulianpascalau) [reply keysManagement/keysHolder.go:204]: not quite an improvement as it will be also a var. `holder.isMainMachine` is already a bool var

**Review by raduchis — APPROVED** [review]

**Pull request merged**